doc/stdenv/passthru.chapter.md

```diff
@@ -94,19 +94,21 @@ stdenv.mkDerivation (finalAttrs: {
   installCheckPhase = ''
     runHook preInstallCheck
     echo checking if 'git --version' mentions ${finalAttrs.version}
-    $out/bin/git --version | grep ${finalAttrs.version}
+    env --ignore-environment $out/bin/git --version | grep --silent "${finalAttrs.version}"
     runHook postInstallCheck
   '';
   # ...
 })
 ```
+
+Testing with a clean environment is encouraged to ensure the `bin/` entrypoints are properly wrapped and self-contained, and do not depend on variables like PYTHONPATH and PERL5LIB being configured.
 :::
 
 However, tests that are non-trivial will better fit into `passthru.tests` because they:
 
-- Access the package as consumers would, independently from the environment in which it was built
+- Access the package outputs as any consumer would, with an environment as setup up with either `stdenv` or `environment.systemPackages`, and without temporary build artifacts and state produced in `configurePhase` and `buildPhase`.
```

Comment:
```suggestion
- Access the package outputs as any consumer would, with an environment set up with either `stdenv` or `environment.systemPackages`, and without temporary build artifacts and state produced in `configurePhase` and `buildPhase`.
```

Just fixes. Generally like this better, though I'm still personally a little unsure about the "with an environment set up with either `stdenv` or `environment.systemPackages`" bit: 
- Is it clarifying something specific beyond what "as any consumer would" already implies?
- I'm not sure I grok how the `environment.systemPackages` bit fits--is there any testing context in which a passthru test would run with all of the user's systemPackages in scope? (That seems undesirable and a sure source of flakiness, unless maybe it's gesturing at grafting in NixOS VM tests? If the latter, it's worth clarifying.)
- It reads like a complete list, but I imagine it's leaving out cases?

Reply:
The reason I mention those two means of consuming a package is because of the various hooks that run through the packages and ensure their propagated inputs are also added  and set up in various ways. It should read as non-exhausive.

The 

how about

```suggestion
- Access the package outputs as any consumer would, for example like the environments set up when using `nix-shell`, `stdenv`, or `environment.systemPackages`, and without the temporary build artifacts and state produced in `configurePhase` and `buildPhase`.
```